packages/mattermost-redux/src/actions/users.ts

```diff
@@ -219,32 +127,45 @@ export function loadMe(): ActionFunc {
             Client4.attachDevice(deviceId);
         }
 
-        const promises = [
-            dispatch(getMe()),
-            dispatch(getMyPreferences()),
-            dispatch(getMyTeams()),
-            dispatch(getMyTeamMembers()),
-        ];
-
         // Sometimes the server version is set in one or the other
         const serverVersion = Client4.getServerVersion() || getState().entities.general.serverVersion;
         dispatch(setServerVersion(serverVersion));
 
-        await Promise.all(promises);
+        const isCollapsedThreads = isCollapsedThreadsEnabled(getState());
+
+        try {
+            await Promise.all([
+                dispatch(getMe()),
+                dispatch(getMyPreferences()),
+                dispatch(getMyTeams()),
+                dispatch(getMyTeamMembers()),
+            ]);
+
+            await dispatch(getMyTeamUnreads(isCollapsedThreads));
+        } catch (error) {
+            if (isLoggedIn) {
```

Comment:
I'm not sure if I like including login logic in this function since it feels like we're mixing up two different things in this function: loading the user, and finishing the login. Maybe we could re-add a version of `completeLogin` that just calls this, waits for it to finish, and then dispatches LOGIN_FAILURE or LOGIN_SUCCESS based on the result? That way, we'll be reusing the same logic for loading everything, but the login handling will be separate from the loading